pkg/controller/deployment/deployment_controller.go

```diff
@@ -1057,14 +1107,30 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 		UpdatedReplicas:     updatedReplicas,
 		AvailableReplicas:   availableReplicas,
 		UnavailableReplicas: unavailableReplicas,
+		EnactedReplicas:     enactedReplicas,
 	}
+	glog.V(4).Infof("Updating status for deployment %q (generation: %d)", deployment.Name, deployment.Generation)
 	_, err = dc.client.Extensions().Deployments(deployment.ObjectMeta.Namespace).UpdateStatus(&newDeployment)
 	return err
 }
 
-func (dc *DeploymentController) calculateStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, deployment extensions.Deployment) (totalActualReplicas, updatedReplicas, availableReplicas, unavailableReplicas int, err error) {
+func (dc *DeploymentController) calculateStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, deployment extensions.Deployment) (totalActualReplicas, updatedReplicas, availableReplicas, unavailableReplicas, enactedReplicas int, err error) {
 	totalActualReplicas = deploymentutil.GetActualReplicaCountForReplicaSets(allRSs)
-	updatedReplicas = deploymentutil.GetActualReplicaCountForReplicaSets([]*extensions.ReplicaSet{newRS})
+	if newRS != nil {
+		updatedReplicas = deploymentutil.GetActualReplicaCountForReplicaSets([]*extensions.ReplicaSet{newRS})
```

Comment:
Should be GetReplicaCountForReplicaSets -- the Spec replicas value, not the Status replicas value.